bonjourserviceregister.cpp

```diff
@@ -34,41 +34,74 @@ ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 #include <QtCore/QSocketNotifier>
 #include <QtCore/QtEndian>
 
-BonjourServiceRegister::BonjourServiceRegister(QObject *parent)
-		: QObject(parent), dnssref(0), bonjourSocket(0) {
+BonjourServiceRegister::BonjourServiceRegister(const BonjourRecord &record, quint16 servicePort, int niIndex, QObject *parent) :
+    QObject(parent)
+  , dnssref(nullptr)
+  , m_record(record)
+  , m_servicePort(servicePort)
+  , m_niIndex(niIndex)
+  , m_thread(nullptr)
+{
+    m_thread = new QThread();
+    this->moveToThread(m_thread);
+
+    connect(m_thread, &QThread::started, this, &BonjourServiceRegister::onStarted);
+    connect(m_thread, &QThread::finished, this, &BonjourServiceRegister::deleteLater);
+    connect(m_thread, &QThread::finished, m_thread, &QThread::deleteLater);
 }
 
-BonjourServiceRegister::~BonjourServiceRegister() {
-	if (dnssref) {
-		DNSServiceRefDeallocate(dnssref);
-		dnssref = 0;
-	}
+BonjourServiceRegister::~BonjourServiceRegister()
+{
+    if (dnssref != nullptr) {
```

Comment:
Ensure to call unregister service when destructor is called.

Reply:
Ok, done